src/Node/ClassPropertyNode.php

```diff
@@ -21,6 +21,7 @@ public function __construct(
 		private ?string $phpDoc,
 		private bool $isPromoted,
 		Node $originalNode,
+		private bool $isReadonlyByPhpDoc = false,
```

Comment:
There isn't a reason for the parameter to be optional

Reply:
Constructors are special :) https://phpstan.org/developing-extensions/backward-compatibility-promise

> Objects of classes with `@api` in their PHPDoc cannot be created via `new` unless there's also `@api` above the public constructor. This encourages using [dependency injection](/developing-extensions/dependency-injection-configuration) in case the class is registered as a service in the dependency injection container.